src/components/markdown-renderer/markdown-it-plugins/quote-extra.ts

```diff
@@ -4,44 +4,100 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
-import MarkdownIt from 'markdown-it'
-import markdownItRegex from 'markdown-it-regex'
-import { RegexOptions } from '../../../external-types/markdown-it-regex/interface'
-
-export const quoteExtra: MarkdownIt.PluginSimple = (markdownIt) => {
-  markdownItRegex(markdownIt, replaceQuoteExtraAuthor)
-  markdownItRegex(markdownIt, replaceQuoteExtraColor)
-  markdownItRegex(markdownIt, replaceQuoteExtraTime)
+import MarkdownIt from 'markdown-it/lib'
+import StateInline from 'markdown-it/lib/rules_inline/state_inline'
+import Token from 'markdown-it/lib/token'
+import { IconName } from '../../common/fork-awesome/types'
+
+export interface QuoteExtraOptions {
+  quoteLabel: string
+  icon: IconName
 }
 
-const replaceQuoteExtraTime: RegexOptions = {
-  name: 'quote-extra-time',
-  regex: /\[time=([^\]]+)]/,
-  replace: (match) => {
-    // ESLint wants to collapse this tag, but then the tag won't be valid html anymore.
-    // noinspection CheckTagEmptyBody
-    return `<span class="quote-extra"><i class="fa fa-clock-o mx-1"></i> ${ match }</span>`
+export type PartParser = (line: string, start: number, maxPos: number) => (number | undefined)
+
+export const quoteExtra: (pluginOptions: QuoteExtraOptions) => MarkdownIt.PluginSimple =
+  (pluginOptions) => (md) => {
+    md.inline.ruler.push(`extraQuote_${ pluginOptions.quoteLabel }`, (state) => {
+      if (state.src[state.pos] !== '[') {
+        return false
+      }
+
+      const quoteLabel = parsePart(state, state.pos + 1, parseLabel)
+      if (!quoteLabel || quoteLabel !== pluginOptions.quoteLabel) {
+        return false
+      }
+
+      const quoteValue = parsePart(state, state.pos + quoteLabel.length + 2, parseValue)
+      if (!quoteValue) {
+        return false
+      }
+
+      state.pos += quoteLabel.length + quoteValue.length + 3
+
+      const tokens: Token[] = []
+      state.md.inline.parse(
+        quoteValue,
+        state.md,
+        state.env,
+        tokens
+      )
+
+      //> [name=ChengHan Wu] [time=Sun, Jun 28, 2015 9:59 PM] [color=#907bf7]
```

Comment:
Should this be included here?

Reply:
This was my demo content. I have commited it by mistake. Removed